syntax = "proto3";

package SaveObject;

message GameItem
{
	uint64 index = 1;
	uint64 num = 2;
}

message TypeInventory
{
	uint32 type = 1;
	
	repeated GameItem items = 2;
}

message Inventory
{
	enum ItemType
	{
		CLOTH = 0;
		WEAPON = 1;
		ITEM = 2;
	}

	repeated TypeInventory typeInventory = 1;
}

message QuestStatus
{
	uint32 index = 1;
	
	enum QuestType
	{
		SERIAL = 0;
		PARALLEL = 1;
	}
	QuestType type = 2;

	enum QuestProgressType
	{
		UNAVAILABLE = 0;
		AVAILABLE = 1;
		INPROGRESS = 2;
		COMPLETABLE = 3;
		COMPLETED = 4;
	}
	QuestProgressType progressType = 3;

	message SubQuestStatus
	{
		enum SubQuestType
		{
			ARRIVAL = 0;
			HUNT = 1;
			ITEM = 2;
			ACTION = 3;
		}
		SubQuestType type = 1;

		bool bStarted = 2;
		bool bCompleted = 3;
	
		uint32 currAmount = 4;
	}
	repeated SubQuestStatus subStatus = 4;

	uint32 currPhase = 5;
	uint32 completed = 6;
}

message PlayerInfo
{
	string name = 1;
	uint32 level = 2;
	uint32 map = 3;

	float loc_x = 4;
	float loc_y = 5;
	float loc_z = 6;

	uint64 gold = 7;

	Inventory inventory = 8;

	repeated QuestStatus questStatus = 9;

	uint32 slotIndex = 10;
}